dll/cpl/timedate/internettime.c

```diff
@@ -177,6 +188,74 @@ EnableDialogText(HWND hwnd)
     EnableWindow(GetDlgItem(hwnd, IDC_NEXTSYNC), bChecked);
 }
 
+static VOID
+SyncNTPStatusInit(HWND hwnd)
+{
+    /* Initialize the Synchronization NTP status members */
+    LoadStringW(hApplet, IDS_INETTIMEWELCOME, SyncStatus.szSyncInit, _countof(SyncStatus.szSyncInit));
+    LoadStringW(hApplet, IDS_INETTIMESUCSYNC, SyncStatus.szSyncSuc, _countof(SyncStatus.szSyncSuc));
+    LoadStringW(hApplet, IDS_INETTIMESYNCING, SyncStatus.szSyncWait, _countof(SyncStatus.szSyncWait));
+    LoadStringW(hApplet, IDS_INETTIMEERROR, SyncStatus.szSyncErr, _countof(SyncStatus.szSyncErr));
+    LoadStringW(hApplet, IDS_INETTIMESUCFILL, SyncStatus.szSyncType, _countof(SyncStatus.szSyncType));
+
+    /*
+     * TODO: XP's and Server 2003's timedate.cpl loads the last successful attempt of the NTP synchronization
+     * displaying the last time and date of the said sync. I have no idea how does timedate.cpl remember its last
+     * successful sync so for the time being, we will only load the initial remark string.
+    */
+    SyncStatus.lpszSyncStatus = SyncStatus.szSyncInit;
+    SetDlgItemTextW(hwnd, IDC_SUCSYNC, SyncStatus.lpszSyncStatus);
+}
+
+static VOID
+UpdateNTPStatus(HWND hwnd, DWORD dwReason)
+{
+    WCHAR szFormat[BUFSIZE];
+    WCHAR szNtpServerName[MAX_VALUE_NAME];
+    HWND hDlgComboList;
+    SYSTEMTIME lt;
+
+    /* Retrieve the server NTP name from the edit box */
+    hDlgComboList = GetDlgItem(hwnd, IDC_SERVERLIST);
+    (UINT)SendMessageW(hDlgComboList, CB_GETCURSEL, 0, 0);
+    SendMessageW(hDlgComboList, WM_GETTEXT, _countof(szNtpServerName), (LPARAM)szNtpServerName);
+
+    /* Iterate over the case reasons so we can compute the exact status of the NTP synchronization */
+    switch (dwReason)
+    {
+        /* The NTP time synchronization has completed successfully */
+        case ERROR_SUCCESS:
+        {
+            /* Retrieve the current time of the system */
+            GetLocalTime(&lt);
+
+            /* Format the resource sting with the given NTP server name and the current time data */
+            StringCchPrintfW(szFormat, _countof(szFormat), SyncStatus.szSyncSuc, szNtpServerName, lt.wDay, lt.wMonth, lt.wYear, lt.wHour, lt.wMinute);
```

Comment:
This type of format doesn't work, unfortunately. Different locales need day/month/year to be in different orders (e.g. en-US is "mm/dd/yyyy", ISO is "yyyy-mm-dd"). So this should use `GetTimeFormatW` or similar.